niceman/resource/session.py

```diff
@@ -194,49 +256,137 @@ def niceman_exec(self, command, args):
 
     @abc.abstractmethod
     def exists(self, path):
-        """Return if file exists"""
-        pass
+        """Determine if a path exists in the resource.
+        
+        Parameters
+        ----------
+        path : string
+            Path to check for existence in the resource environment.
+        
+        Returns
+        -------
+        bool
+            True if path exists
+        """
+        raise NotImplementedError
 
     @abc.abstractmethod
-    def put(self, src_path, dest_path, preserve_perms=False,
-                owner=None, group=None, recursive=False):
-        """Take file on the local file system and copy over into the session
+    def put(self, src_path, dest_path, uid=-1, gid=-1):
```

Comment:
eh heh -- so we will stick to C / `os.chown` agreement on `-1` to state "do nothing"... ok for now